### ISSUE TITLE
BgzfWriter does not check that a fileobj is writeable

### ISSUE THREAD

**Issue by fabianegli** [biopython/biopython#3751]
### Expected behaviour

BgzfWriter checks that a fileobj is writeable.

### Actual behaviour

BgzfWriter does not check that a fileobj is writeable.

### Steps to reproduce

```
>>> from Bio import bgzf
>>> bgzf.BgzfWriter(fileobj=open("./Tests/Quality/example.fastq.bgz", "rb"))
<Bio.bgzf.BgzfWriter object at 0x101a24bb0>
```

**Comment by peterjc** [comment]
Good point, it will eventually raise ``io.UnsupportedOperation: write`` once the the first buffered block is full, or it is flushed or closed.

Perhaps this the best approach (which will also verify the handle will accept bytes)?

```python
try:
    fileobj.write(b"")
except (TypeError, OSError):
    raise ValueError("fileobj must be an output handle in binary mode"
```

Related to this, trying to use ``BgzfReader`` and an output handle we immediately get ``io.UnsupportedOperation: read`` which is fairly clear, so the is less value in adding a check.
